example/lib/main.dart

```diff
@@ -29,37 +30,67 @@ class MyApp extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
     return MaterialApp(
+      theme: ThemeData(
+        elevatedButtonTheme: ElevatedButtonThemeData(
+          style: ButtonStyle(
+            padding: MaterialStateProperty.all(const EdgeInsets.all(20)),
+          ),
+        ),
+      ),
       home: Scaffold(
         body: Builder(
           builder: (context) {
-            return Column(
-              children: [
-                ElevatedButton(
-                  child: const Text('Sign in with google'),
-                  onPressed: signInWithArgs(
-                    context,
-                    GoogleSignInArgs(
-                      clientId:
-                          '448618578101-sg12d2qin42cpr00f8b0gehs5s7inm0v.apps.googleusercontent.com',
-                      redirectUri:
-                          'https://react-native-firebase-testing.firebaseapp.com/__/auth/handler',
-                    ),
+            final buttons = [
+              ElevatedButton(
+                child: const Text('Sign in with Google'),
+                onPressed: signInWithArgs(
+                  context,
+                  GoogleSignInArgs(
+                    clientId:
+                        '448618578101-sg12d2qin42cpr00f8b0gehs5s7inm0v.apps.googleusercontent.com',
+                    redirectUri:
```

Comment:
All providers have the same redirectUri?

Reply:
yes.
This is up to the user to configure. Existing oauth example apps from flutterfire repo are configured with this callback url.
A bit of context: if that would have been a backend thing – unique path for each provider would have been crucial, but since it's client side logic, depending on path parameter is not necessary, since we already know which provider flow is in currently in action